### PR TITLE
Move to docc documentation

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/apple/swift-metrics-extras/issues/18

### CODE REVIEW

**Comment by ktoso** [comment]
The nightly failure is somethign we must fix independently: https://github.com/apple/swift-metrics-extras/issues/17

File: Sources/SystemMetrics/Docs.docc/index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+```swift
+import SystemMetrics
+```
+
+This makes the System Metrics API available. This adds a new method to `MetricsSystem` called `bootstrapWithSystemMetrics`. Calling this method will call `MetricsSystem.bootstrap` as well as bootstrapping System Metrics.
+
+`bootstrapWithSystemMetrics` takes a `SystemMetrics.Configuration` object to configure the system metrics. The config has the following properties:
+
+- interval: The interval at which SystemMetrics are being calculated & exported.
```
Comment (yim-lee) [inline Sources/SystemMetrics/Docs.docc/index.md:30]:
```suggestion
- interval: The interval at which `SystemMetrics` are being calculated & exported.
```

?

Or System Metrics?

File: Sources/SystemMetrics/Docs.docc/index.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- dataProvider: A closure returing `SystemMetrics.Data?`. When `nil` no metrics are exported (the default on non-linux platforms). `SystemMetrics.Data` holds all the values mentioned above.
```
Comment (yim-lee) [inline Sources/SystemMetrics/Docs.docc/index.md:31]:
```suggestion
- dataProvider: A closure returning `SystemMetrics.Data?`. When `nil`, no metrics are exported (the default on non-Linux platforms). `SystemMetrics.Data` holds all the values mentioned above.
```

File: docker/docker-compose.2004.56.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  runtime-setup:
+    image: swift-metrics:20.04-5.6
+    build:
+      args:
+        ubuntu_version: "focal"
+        swift_version: "5.6"
+
+  test:
+    image: swift-metrics:20.04-5.6∂
```
Comment (yim-lee) [inline docker/docker-compose.2004.56.yaml:13]:
```suggestion
    image: swift-metrics:20.04-5.6
```

it looks pretty cool though :)

File: docker/docker-compose.2004.57.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+version: "3"
+
+services:
+
+  runtime-setup:
+    image: swift-metrics:20.04-5.7
+    build:
+      args:
+        ubuntu_version: "focal"
+        swift_version: "5.7"
```
Comment (yim-lee) [inline docker/docker-compose.2004.57.yaml:10]:
```suggestion
        base_image: "swiftlang/swift:nightly-5.7-focal"
```

File: scripts/docs/preview_docc.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift Distributed Actors project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+xcrun swift package --disable-sandbox preview-documentation --target $1
```
Comment (yim-lee) [inline scripts/docs/preview_docc.sh:16]: nice. good idea to add this to other repos 👍

File: scripts/soundness.sh [hunk 6]
```diff
@@ -88,7 +88,7 @@ for language in swift-or-c bash dtrace; do
   matching_files=( -name '*' )
   case "$language" in
       swift-or-c)
-        exceptions=( -name c_nio_http_parser.c -o -name c_nio_http_parser.h -o -name cpp_magic.h -o -name Package.swift -o -name CNIOSHA1.h -o -name c_nio_sha1.c -o -name ifaddrs-android.c -o -name ifaddrs-android.h)
+        exceptions=( -name c_nio_http_parser.c -o -name c_nio_http_parser.h -o -name cpp_magic.h -o -name Package@*.swift -o -name CNIOSHA1.h -o -name c_nio_sha1.c -o -name ifaddrs-android.c -o -name ifaddrs-android.h)
```
Comment (yim-lee) [inline scripts/soundness.sh:91]: Maybe clean up the list and remove those that don't apply to this repo?

**Review by yim-lee — APPROVED** [review]
Some minor suggestions. LGTM overall. 

Perhaps should wait for the outcome of the discussion around `Package.swift` in https://github.com/apple/swift-statsd-client/pull/36 before merging.

**Comment by yim-lee** [comment]
Added CI for Swift 5.6 and 5.7. 

@swift-server-bot test this please

**Comment by ktoso** [comment]
Okey I chimed in on https://github.com/apple/swift-statsd-client/pull/36 

THanks for review!

File: scripts/docs/preview_docc.sh (hunk 5 above)
Comment (ktoso) [reply scripts/docs/preview_docc.sh:16]: :-) I find it helpful, as the command is quite terrible tbh...

File: Sources/SystemMetrics/Docs.docc/index.md (hunk 1 above)
Comment (ktoso) [reply Sources/SystemMetrics/Docs.docc/index.md:30]: probably ok like this :)

File: docker/docker-compose.2004.56.yaml (hunk 3 above)
Comment (ktoso) [reply docker/docker-compose.2004.56.yaml:13]: oh lol, thanks :)

File: scripts/soundness.sh (hunk 6 above)
Comment (ktoso) [reply scripts/soundness.sh:91]: okey will do

File: README.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ or other such metric system agnostic metrics additions–such additions are perf
 To add a dependency on the extras package, you need to declare it in your `Package.swift`:
 
 ```swift
-.package(url: "https://github.com/apple/swift-metrics-extras.git", "1.0.0" ..< "2.0.0"),
+.package(url: "https://github.com/apple/swift-metrics-extras.git", "0.1.0" ..< "1.0.0"),
```
Comment (tomerd) [inline README.md:25]: probably should be `from:` ?

Comment (ktoso) [reply README.md:25]: Nice catch, thanks

File: scripts/docs/preview_docc.sh [hunk 8]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
```
[the remark this answers is not in the document]
Comment (tomerd) [reply scripts/docs/preview_docc.sh:2]: using Swift Metric is fine, but this file has `Swift Distributed Actors` which should be fixed

File: docker/Dockerfile [hunk 9]
```diff
@@ -16,11 +16,8 @@ ENV LANGUAGE en_US.UTF-8
 RUN apt-get update && apt-get install -y wget
 RUN apt-get update && apt-get install -y lsof dnsutils netcat-openbsd net-tools curl jq # used by integration tests
 
-# ruby and jazzy for docs generation
+# ruby
 RUN apt-get update && apt-get install -y ruby ruby-dev libsqlite3-dev build-essential
-# jazzy no longer works on xenial as ruby is too old.
-RUN if [ "${ubuntu_version}" = "focal" ] ; then echo "gem: --no-document" > ~/.gemrc; fi
-RUN if [ "${ubuntu_version}" = "focal" ] ; then gem install jazzy; fi
```
Comment (tomerd) [inline docker/Dockerfile:23]: 🎉

File: scripts/docs/preview_docc.sh (hunk 8 above)
[the remark this answers is not in the document]
Comment (ktoso) [reply scripts/docs/preview_docc.sh:2]: whoops fixed

File: Sources/SystemMetrics/Docs.docc/index.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+After adding `swift-metrics-extras` as a dependency you can import the `SystemMetrics` module.
+
+```swift
+import SystemMetrics
+```
+
+This makes the System Metrics API available. This adds a new method to `MetricsSystem` called `bootstrapWithSystemMetrics`. Calling this method will call `MetricsSystem.bootstrap` as well as bootstrapping System Metrics.
+
+`bootstrapWithSystemMetrics` takes a `SystemMetrics.Configuration` object to configure the system metrics. The config has the following properties:
```
[the remark this answers is not in the document]
Comment (ktoso) [reply Sources/SystemMetrics/Docs.docc/index.md:28]: Sadly not across modules :-(

File: scripts/soundness.sh [hunk 11]
```diff
@@ -88,7 +88,7 @@ for language in swift-or-c bash dtrace; do
   matching_files=( -name '*' )
   case "$language" in
       swift-or-c)
-        exceptions=( -name c_nio_http_parser.c -o -name c_nio_http_parser.h -o -name cpp_magic.h -o -name Package.swift -o -name CNIOSHA1.h -o -name c_nio_sha1.c -o -name ifaddrs-android.c -o -name ifaddrs-android.h)
+        exceptions=( -name Package.swift -o -name Package@*.swift )
```
Comment (yim-lee) [inline scripts/soundness.sh:91]: I think we need quotes around `Package@*.swift`.

Comment (ktoso) [reply scripts/soundness.sh:91]: thanks!

**Pull request merged**